### PR TITLE
doc: Update metrics documentation & list exported metrics

### PR DESCRIPTION
**Pull request by raybejjani**
We export a small list of metrics and now list them in the docs. This
change also includes some text fixups and a bad link to the prometheus
configuration.

Relates to: https://github.com/cilium/cilium/issues/3597

### CODE REVIEW

**Comment by raybejjani** [comment]
test-me-please

**Comment by borkmann** [comment]
test-me-please

File: Documentation/configuration/metrics.rst [hunk 1]
```diff
@@ -4,43 +4,73 @@
...
+  source: ``api``, ``containerd``, ``k8s``.
+
+Cilium as a kubernetes pod
 ==========================
-The Prometheus reference configuration includes "jobs" to automatically collect pod metrics marked appropriately. Your cilium spec will need two labels:
+The Cilium `Prometheus reference configuration <https://github.com/cilium/cilium/blob/master/examples/kubernetes/prometheus.yaml>`_
+configures jobs that automatically collect pod metrics marked with the
+appropriate two labels.
+
+Your Cilium spec will need these labels:
```
Comment (tgraf) [inline Documentation/configuration/metrics.rst]: Annotations?

**Review by tgraf — CHANGES_REQUESTED** [review]

File: Documentation/configuration/metrics.rst [hunk 2]
```diff
@@ -4,43 +4,73 @@
...
+* ``policy_count``: Number of policies currently loaded
+* ``policy_max_revision``: Highest policy revision number in the agent
+* ``policy_import_errors``: Number of times a policy import has failed
+
+Events external to Cilium
+-------------------------
+* ``event_ts``: Last timestamp when we received an event. Further labelled by
+  source: ``api``, ``containerd``, ``k8s``.
+
+Cilium as a kubernetes pod
```
Comment (ianvernon) [inline Documentation/configuration/metrics.rst]: kubernetes --> Kubernetes

File: Documentation/configuration/metrics.rst [hunk 3]
```diff
@@ -4,43 +4,73 @@
...
+*Note: the port can be configured per-pod to any value and the label set
+accordingly. Prometheus uses this label to discover the port.*
 
-An example of how to do this can be found in the cilium
-`kubernetes example spec file <https://github.com/cilium/cilium/blob/master/examples/kubernetes/cilium.yaml>`_
+To configure automatic metric discovery and collection, Prometheus itself requires a
+`kubernetes_sd_config configuration <https://prometheus.io/docs/prometheus/latest/configuration/configuration/>`_.
+The configured rules are used to filter pods and nodes by label and
+annotation, and tag the resulting metrics series. In the kubernetes case
+Prometheus will contact the kubernetes API server for these lists and must have
```
Comment (ianvernon) [inline Documentation/configuration/metrics.rst]: kubernetes --> Kubernetes

File: Documentation/configuration/metrics.rst [hunk 4]
```diff
@@ -4,43 +4,73 @@
...
+
+*Note: the port can be configured per-pod to any value and the label set
+accordingly. Prometheus uses this label to discover the port.*
 
-An example of how to do this can be found in the cilium
-`kubernetes example spec file <https://github.com/cilium/cilium/blob/master/examples/kubernetes/cilium.yaml>`_
+To configure automatic metric discovery and collection, Prometheus itself requires a
+`kubernetes_sd_config configuration <https://prometheus.io/docs/prometheus/latest/configuration/configuration/>`_.
+The configured rules are used to filter pods and nodes by label and
+annotation, and tag the resulting metrics series. In the kubernetes case
```
Comment (ianvernon) [inline Documentation/configuration/metrics.rst]: kubernetes --> Kubernetes

File: Documentation/configuration/metrics.rst [hunk 5]
```diff
@@ -77,17 +107,18 @@ by the kubernetes API server:
 
 - find and keep all pods that have labels ``k8s-app=cilium`` and ``prometheus.io/scrape=true``
 - extract the IP and port of the pod from ``address`` and ``prometheus.io/port``
-- discover the metrics url path from the label ``prometheus.io/path`` and uses the default of ``/metrics`` when it isn't present
+- discover the metrics url path from the label ``prometheus.io/path`` or use the default of ``/metrics`` when it isn't present
```
Comment (ianvernon) [inline Documentation/configuration/metrics.rst]: url --> URL

**Review by ianvernon — CHANGES_REQUESTED** [review]
Small nits.

**Comment by raybejjani** [comment]
test-me-please

**Review by tgraf — APPROVED** [review]

**Pull request merged**